certstore_windows: reimplement without cgo

sailorfrag opened the pull request.

sailorfrag committed:
certstore_windows: reimplement without cgo

sailorfrag commented:
Regarding the CI failures: it appears that these were failing already (not sure if this is because the CI runner configuration changed over time or the `cyolosecurity` fork maybe broke something) but we don't use the `Delete` functionality anyway. I'd like to figure that out at some point, but perhaps not today.

sailorfrag commented:
Also, I was tempted to simplify and/or refactor various bits but resisted it to make sure that the before/after diff reads more like a transliteration rather than a rewrite.

dblohm7 commented on syscall_windows.go:
```diff
@@ -0,0 +1,82 @@
+// https://learn.microsoft.com/en-us/windows/win32/api/ncrypt/nf-ncrypt-ncryptsignhash
+//sys nCryptSignHash(hKey nCryptKeyHandle, pPaddingInfo unsafe.Pointer, pbHashValue *byte, cbHashValue uint32, pbSignature *byte, cbSignature uint32, pcbResult *uint32, dwFlags uint32) (ss securityStatus) = ncrypt.NCryptSignHash
+// https://learn.microsoft.com/en-us/windows/win32/api/ncrypt/nf-ncrypt-ncryptdeletekey
+//sys nCryptDeleteKey(hKey nCryptKeyHandle, dwFlags uint32) (ss securityStatus) = ncrypt.NCryptDeleteKey
```
Instead of using `ss securityStatus` as the return type here, you can use `ret error` (as opposed to `err error`) and the code generator will do what you want, then you can remove the definition of `securityStatus` entirely.

sailorfrag replied:
Oh that's much better

dblohm7 reviewed (changes requested):
I'd just like to see this once more.

sailorfrag commented:
I managed to fix the tests. `CryptGetProvParam` queries for `PP_CONTAINER` and `PP_NAME` return ASCII/ANSI/whatever byte strings, and `windows.CryptAcquireContext` calls `CryptAcquireContextW` which expects UTF-16 strings. I didn't find a way to query for UTF-16 names, so I added a conversion.

I couldn't run the tests locally because they shell out to `openssl` to perform PFX encoding and I couldn't get that to work on my test system. I switched that to use an imported package instead.
